enterprise_access/apps/api/v1/views/content_assignments/assignments.py

```diff
@@ -0,0 +1,87 @@
+"""
+Learner-facing REST API views for LearnerContentAssignments in the content_assignments app.
+"""
+import logging
+
+from drf_spectacular.utils import extend_schema
+from edx_rbac.decorators import permission_required
+from edx_rest_framework_extensions.auth.jwt.authentication import JwtAuthentication
+from rest_framework import authentication, mixins, permissions, status, viewsets
+
+from enterprise_access.apps.api import filters, serializers
+from enterprise_access.apps.api.v1.views.utils import PaginationWithPageCount
+from enterprise_access.apps.content_assignments.models import LearnerContentAssignment
+from enterprise_access.apps.core.constants import CONTENT_ASSIGNMENT_LEARNER_READ_PERMISSION
+
+logger = logging.getLogger(__name__)
+
+CONTENT_ASSIGNMENT_CRUD_API_TAG = 'Content Assignment CRUD'
+
+
+class LearnerContentAssignmentViewSet(
+    mixins.ListModelMixin,
+    mixins.RetrieveModelMixin,
+    viewsets.GenericViewSet,
+):
+    """
+    Viewset supporting all Admin CRUD operations on ``LearnerContentAssignment`` records.
+    """
+    permission_classes = (permissions.IsAuthenticated,)
+    serializer_class = serializers.LearnerContentAssignmentResponseSerializer
+    authentication_classes = (JwtAuthentication, authentication.SessionAuthentication)
+    filter_backends = (filters.NoFilterOnDetailBackend,)
+    filterset_class = filters.AssignmentFilter
```

Comment:
Do we need filtering on the learner version?  And if so, can learners filter on the same fields as admins?

Reply:
Right, at least not the same as admins.  The one's the learners might conceivably every want to filter on are content_key and state.  I don't know if the learner portal would ever need to do that filtering though.

Reply:
Maybe we just not pre-emtively introduce dead code and not allow learners to filter the list results.

Reply:
Either one sounds reasonable to me!

Reply:
anyway, this is my only concern, I'll approve so you can merge at your convenience